salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls

```diff
@@ -0,0 +1,31 @@
+{% set hostname = salt['network.get_hostname']() %}
+
+{% set kubeconfig = "/etc/kubernetes/admin.conf" %}
+{% set context = "kubernetes-admin@kubernetes" %}
+
+Apply control-plane role label:
+  kubernetes.node_label_present:
+    - name: "node-role.kubernetes.io/master"
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - node: {{ hostname }}
+    - value: ""
+
+Apply control-plane taints:
+  kubernetes.node_taints_present:
+    - name: {{ hostname }}
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - taints:
+        - key: "node-role.kubernetes.io/master"
+          effect: "NoSchedule"
+
+Annotate with CRI socket information:
+  kubernetes.node_annotation_present:
+    - name: "kubeadm.alpha.kubernetes.io/cri-socket"
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - node: {{ hostname }}
+    # TODO: the socket location may become configurable, we should make sure
```

Comment:
I think, it's already configurable in default.yaml: `kubelet.service.options.container-runtime-endpoint`